osu.Game/Overlays/Dashboard/CurrentlyPlayingDisplay.cs

```diff
@@ -57,6 +100,57 @@ protected override void LoadComplete()
             playingUsers.BindCollectionChanged(onPlayingUsersChanged, true);
         }
 
+        private void addCard(int userId)
+        {
+            if (currentUsers.Contains(userId))
+                return;
+
+            users.GetUserAsync(userId).ContinueWith(task =>
+            {
+                var user = task.GetResultSafely();
+
+                if (user == null)
+                    return;
+
+                if (!user.Username.ToLower().Contains(searchBar.Text.ToLower()))
+                    return;
+
+                Schedule(() =>
+                {
+                    // user may no longer be playing.
+                    if (!playingUsers.Contains(user.Id))
+                        return;
+
+                    currentUsers.Add(user.Id);
+
+                    userFlow.Add(createUserPanel(user));
+                });
+            });
+        }
+
+        private void removeCard(PlayingUserPanel card)
+        {
+            if (card == null)
+                return;
+
+            currentUsers.Remove(card.User.Id);
+            card.Expire();
+        }
+
+        private void onSearchBarValueChanged(ValueChangedEvent<string> change)
+        {
+            foreach (PlayingUserPanel card in userFlow)
+            {
+                if (!card.User.Username.ToLower().Contains(change.NewValue.ToLower()))
+                    removeCard(card);
+            }
+
+            foreach (int userId in playingUsers)
+            {
+                addCard(userId);
+            }
```

Comment:
You may want to take a look at using `SearchContainer` instead which has a better way of filtering drawables (see settings filtering logic for example), rather than manually removing/adding player cards.

Reply:
Done!